src/Stratis.Bitcoin/Consensus/BestChainIndexer.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+
+namespace Stratis.Bitcoin.Consensus
+{
+    /// <summary>
+    /// An indexer of the consensus chain.
+    /// </summary>
+    public class BestChainIndexer
+    {
+        private readonly IConsensusManager consensusManager;
+        private readonly Dictionary<int, ChainedHeader> blocksByHeight = new Dictionary<int, ChainedHeader>();
+        private readonly object lockObj = new object();
```

Comment:
Please add a comment that will mention which objects are protected by the lock. 
Also next to each declaration of protected objects add a comment which will contain "access should be pretected by <see=lockobj> lock".